src/charm.py

```diff
@@ -184,6 +213,15 @@ def _on_database_changed(self, event: DatabaseCreatedEvent) -> None:
         self.unit.status = MaintenanceStatus("Retrieving database details")
         self._update_container(event)
 
+    def _on_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
+        if self.unit.is_leader():
+            type = "admin" if event.relation_name == self.admin_ingress.relation_name else "public"
```

Comment:
`type` is not a great variable name since it is also a python built-in function. Could use `ingress_type`.